### ISSUE TITLE
Warn when the penultimate guard is a not

### ISSUE THREAD

**Issue by josephcsible** [ndmitchell/hlint#835]
When the last guard before an `otherwise` is wrapped in a `not`, we should recommend that they be swapped to avoid the `not`. For example, this:
```haskell
f x
 | x < 2 = doSpecial x
 | not (isPrime x) = doComposite x
 | otherwise = doPrime x
```
Should become this:
```haskell
f x
 | x < 2 = doSpecial x
 | isPrime x = doPrime x
 | otherwise = doComposite x
```

**Comment by ndmitchell** [comment]
I think sometimes its fine to swap the order of guards, and sometimes people have chosen then very carefully. It's why we don't suggest `if not c then t else f ==> if c then f else t`, so I'd be careful with this warning.

**Comment by josephcsible** [comment]
I didn't realize we didn't have that `if` hint, or I actually would have PR'd it. I don't see when you'd want to choose to have the `not`. Can you give an example?

**Comment by ndmitchell** [comment]
Sometimes the else branch is huge and then if is small, so it makes sense to put the small one first to get it out the way, e.g.:

```
timed c i x = if not useTimings then x else unsafePerformIO $ timedIO c i $ evaluate x
```

Sometimes the order of cases is just because that's the logical way to consider them, treating either exceptions or general cases first, but based on how the reader should be expected to think about.
